frontend/packages/noobaa-storage-plugin/src/components/object-bucket-claim-page/modals/attach-deployment-obc-modal.tsx

```diff
@@ -0,0 +1,179 @@
+import * as _ from 'lodash';
+import * as React from 'react';
+import { connect } from 'react-redux';
+import {
+  createModalLauncher,
+  ModalBody,
+  ModalComponentProps,
+  ModalTitle,
+  ModalSubmitFooter,
+} from '@console/internal/components/factory/modal';
+import {
+  Dropdown,
+  history,
+  HandlePromiseProps,
+  resourceObjPath,
+  withHandlePromise,
+  FirehoseResult,
+  Firehose,
+} from '@console/internal/components/utils';
+import { DeploymentModel } from '@console/internal/models';
+import {
+  ContainerSpec,
+  K8sKind,
+  k8sPatch,
+  K8sResourceKind,
+  referenceFor,
+} from '@console/internal/module/k8s/';
+import { getName } from '@console/shared';
+
+const AttachDeploymentToOBCModal = withHandlePromise((props: AttachDeploymentToOBCModalProps) => {
+  const [requestDeployment, setRequestedDeployment] = React.useState({});
+  const [deploymentObjects, setDeployments] = React.useState({});
+  const [deploymentNames, setDeploymentNames] = React.useState({});
+  const { handlePromise, close, cancel, resource, deployments } = props;
+
+  const obcName = getName(resource);
+  const deploymentData = _.get(deployments, 'data');
+  const inProgress = _.get(props, 'loaded');
+  const errorMessage = _.get(props, 'loadError');
+
+  React.useEffect(() => {
+    const deploymentObjectList = {};
+    const deploymentNameList = {};
+
+    _.map(deploymentData, (data) => {
+      const name = getName(data);
+      deploymentObjectList[name] = data;
+      deploymentNameList[name] = name;
+    });
+
+    setDeployments(deploymentObjectList);
+    setDeploymentNames(deploymentNameList);
+  }, [deploymentData, deployments]);
+
+  const getPatches = () => {
+    const configMapRef = {
+      name: 'CONFIGMAP_BUCKET',
+      valueFrom: {
+        configMapKeyRef: {
+          name: obcName,
+          key: obcName,
+        },
+      },
+    };
+    const secretMapRef = {
+      name: 'SECRET_BUCKET',
+      valueFrom: {
+        secretKeyRef: {
+          name: obcName,
+          key: obcName,
+        },
+      },
+    };
+
+    const containers: ContainerSpec[] = _.get(
+      requestDeployment,
+      'spec.template.spec.containers',
+      [],
+    );
+    const patches = containers.reduce((patch, container, i) => {
+      if (_.isEmpty(container.env)) {
+        patch.push({
+          op: 'add',
+          path: `/spec/template/spec/containers/${i}/env`,
+          value: [configMapRef],
+        });
+        patch.push({
+          op: 'add',
+          path: `/spec/template/spec/containers/${i}/env/-`,
+          value: secretMapRef,
+        });
+      } else {
+        patch.push({
+          op: 'add',
+          path: `/spec/template/spec/containers/${i}/env/-`,
+          value: configMapRef,
+        });
+        patch.push({
+          op: 'add',
+          path: `/spec/template/spec/containers/${i}/env/-`,
```

Comment:
Since the path value is getting used again and again. Can we move this into a constant

Reply:
not possible

Reply:
Can you explain with the reason?

Reply:
These are string literals. Moving them out of here and defining them as const would cause ${i} to throw an error. A possible solution would be to break it into three parts and use two const but that would just make things more complex and the part after ${i} is different.